analytics/lib/counts.py

```diff
@@ -193,28 +193,26 @@ def do_aggregate_to_summary_table(stat: CountStat, end_time: datetime,
         logger.info("%s RealmCount aggregation (%dms/%sr)" % (
             stat.property, (end - start) * 1000, cursor.rowcount))
 
-    if realm is None:
-        # Aggregate into InstallationCount.  Only run if we just
-        # processed counts for all realms.
-        #
-        # TODO: Add support for updating installation data after
-        # changing an individual realm's values.
-        installationcount_query = """
-            INSERT INTO analytics_installationcount
-                (value, property, subgroup, end_time)
-            SELECT
-                sum(value), '%(property)s', analytics_realmcount.subgroup, %%(end_time)s
-            FROM analytics_realmcount
-            WHERE
-                property = '%(property)s' AND
-                end_time = %%(end_time)s
-            GROUP BY analytics_realmcount.subgroup
-            """ % {'property': stat.property}
-        start = time.time()
-        cursor.execute(installationcount_query, {'end_time': end_time})
-        end = time.time()
-        logger.info("%s InstallationCount aggregation (%dms/%sr)" % (
-            stat.property, (end - start) * 1000, cursor.rowcount))
+    installationcount_query = """
+        INSERT INTO analytics_installationcount
+            (value, property, subgroup, end_time)
+        SELECT
+            sum(value), '%(property)s', analytics_realmcount.subgroup, %%(end_time)s
+        FROM analytics_realmcount
+        WHERE
+            property = '%(property)s' AND
+            end_time = %%(end_time)s
+        GROUP BY analytics_realmcount.subgroup
+        ON CONFLICT
+            (property, subgroup, end_time)
+        DO UPDATE SET
+            value = EXCLUDED.value
```

Comment:
@arpit551 This was the correct approach, but it doesn't work now that you'd fixed [#13486](https://github.com/zulip/zulip/issues/13486) in https://github.com/zulip/zulip/commit/f299f31340ee36c8af700f9715c7e3001eebfb98 (as there are now two unique constraints where `ON CONFLICT` can handle only one).
Do you have any other suggestions for this?

One approach which can help us fix this is to use `COALESCE`, you can read more about it here https://chat.zulip.org/#narrow/stream/49-development-help/topic/updating.20installation.20count.20table, do tell us what are your thoughts on this?